From: Jun Yang <jun.yang@suse.com>
Date: Thu, 20 Jan 2022 14:52:20 +0800
Subject: [PATCH] spraid: fix compile issue of spraid_main.c
Patch-mainline: Never, fixes out-of-tree driver
References: bsn#71

mainline commit 464a00c9e0ad45e3f42ff6ea705491a356df818e
(scsi: core: Kill DRIVER_SENSE) replace the check for 
DRIVER_SENSE with a check for scsi_status_is_check_condition().

fix build error in spraid driver follow this commit.


Signed-off-by: Jun Yang <jun.yang@suse.com>
---
 drivers/scsi/spraid/spraid_main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/scsi/spraid/spraid_main.c b/drivers/scsi/spraid/spraid_main.c
index c6d2a0b8e35e..9349daec5cc2 100644
--- a/drivers/scsi/spraid/spraid_main.c
+++ b/drivers/scsi/spraid/spraid_main.c
@@ -800,7 +800,7 @@ static void spraid_map_status(struct spraid_iod *iod, struct scsi_cmnd *scmd,
 		if (scmd->result & SAM_STAT_CHECK_CONDITION) {
 			memset(scmd->sense_buffer, 0, SCSI_SENSE_BUFFERSIZE);
 			memcpy(scmd->sense_buffer, iod->sense, SCSI_SENSE_BUFFERSIZE);
-			scmd->result = (scmd->result & 0x00ffffff) | (DRIVER_SENSE << 24);
+			scmd->result = SAM_STAT_CHECK_CONDITION;
 		}
 		break;
 	case FW_STAT_ABORTED:
-- 
2.31.1

